library/kotlin/io/envoyproxy/envoymobile/RequestBuilder.kt

```diff
@@ -0,0 +1,90 @@
+package io.envoyproxy.envoymobile
+
+import java.net.URL
+
+
+/**
+ * Builder used for constructing instances of `Request` types.
+ *
+ * @param url URL for the request.
+ * @param method Method for the request.
+ */
+class RequestBuilder(
+    val url: URL,
```

Comment:
Do we want to support passing `scheme`, `authority`/`host`, and `path` as separate parameters?

Reply:
Maybe phrasing it a little less open-ended: I can think of a couple reasons we might consider supporting it.

The builder could support both options, of course.

Internally (in the request object), I think we should have a split representation.
- Actual protocols handle these attributes entirely separately.
- There are cases where the separate values could encode more information than the combined url.
- You can still re-compose the url if you want it back from the request.

Reply:
@buildbreaker and I had discussed this offline as a consideration. A few thoughts:

- `URL` is required when initializing the builders right now, as is the `RequestMethod`, in order to avoid having `build()` be throwable. We could potentially add another initializer that takes `scheme`/`host`/`path` separately if we want to maintain this and store those values separately under the hood
- I don't think adding methods such as `addHost`/`addScheme` really make sense here since all 3 need to be set to create a valid request - so we'd likely have to pass all of them at once (which could work)

Essentially, I'm okay with storing the 3 values separately, the question is more along the lines of how we want to expose it (my suggestion would be another initializer that takes all 3 in place of the `URL`)

Reply:
Makes sense. That sounds good to me. 👍

Reply:
We can always revisit this in the future. I think having just this right now is a fine solution

Reply:
https://envoyproxy.slack.com/archives/CKQ2LK23G/p1562880255143900